crates/rpc-client/src/contract.rs

```diff
@@ -0,0 +1,250 @@
+use std::collections::HashMap;
+use std::sync::Arc;
+use std::time::Instant;
+
+use anyhow::{anyhow, bail, Result};
+use arc_swap::{ArcSwap, Guard};
+use async_jsonrpc_client::{HttpClient, Params as ClientParams, Transport};
+use gw_config::{BlockProducerConfig, ContractTypeScriptConfig, ContractsCellDep};
+use gw_jsonrpc_types::blockchain::{CellDep, Script};
+use gw_types::packed::RollupConfig;
+use gw_types::prelude::Pack;
+use serde_json::json;
+
+use crate::indexer_types::{Cell, Order, Pagination, ScriptType, SearchKey};
+use crate::rpc_client::RPCClient;
+use crate::utils::to_result;
+
+// Used in block producer and challenge
+#[derive(Clone)]
+pub struct ContractsCellDepManager {
+    rpc_client: RPCClient,
+    scripts: Arc<ContractTypeScriptConfig>,
+    deps: Arc<ArcSwap<ContractsCellDep>>,
```

Comment:
This Arc can also get removed. Elaborately **load()** and **store()** get auto deref twice here. I guess ArcSwap guards the pointee with AtoimicPointer with Strategy internally. So Arc takes no effect here?

```suggestion
    deps:ArcSwap<ContractsCellDep>,
```
This should be fine.

Reply:
No. ArcSwap doesn't implement Clone.